src/app/dataset-block/metadata-block/components/block-navigation/block-navigation.component.ts

```diff
@@ -15,6 +16,7 @@ export class BlockNavigationComponent {
     @Input() public datasetHistory: MaybeNull<DatasetHistoryUpdate>;
     @Input() public currentBlockHash: string;
     @Input() public datasetInfo: DatasetInfo;
+    @Input() public allHistory: MetadataBlockFragment[];
```

Comment:
I suggest to initialize this parameter with an empty array, it will help avoid error if for some reason the parameter will be undefined or null.
```suggestion
    @Input() public allHistory: MetadataBlockFragment[] = [];
```